common/src/main/java/net/william278/huskhomes/redis/redisdata/RoundRobinConnectionPool.java

```diff
@@ -0,0 +1,40 @@
+package net.william278.huskhomes.redis.redisdata;
+
+
+import io.lettuce.core.api.StatefulRedisConnection;
+
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+public class RoundRobinConnectionPool<K,V> {
+    private final AtomicInteger next = new AtomicInteger(0);
+    private StatefulRedisConnection<K,V>[] elements;
+    private final Supplier<StatefulRedisConnection<K,V>> statefulRedisConnectionSupplier;
+
+    public RoundRobinConnectionPool(Supplier<StatefulRedisConnection<K,V>> statefulRedisConnectionSupplier, int poolSize) {
+        this.statefulRedisConnectionSupplier = statefulRedisConnectionSupplier;
+        this.elements = new StatefulRedisConnection[poolSize];
+        for(int i = 0; i < poolSize; i++) {
+            elements[i] = statefulRedisConnectionSupplier.get();
+        }
+    }
+    public void expandPool(int expandBy) {
+        if(expandBy <= 0)
+            throw new IllegalArgumentException("expandBy must be greater than 0");
```

Comment:
style: I prefer fully braced {} if statements in general here -- better for readability and avoiding bugs